codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -716,8 +710,11 @@ let actionReducer = (state: t, action: action) => {
 }
 
 let invalidatedActionReducer = (state: t, action: action) =>
-  switch action {
-  | EventBatchProcessed(_) => ({...state, currentlyProcessingBatch: false}, [Rollback])
+  switch (state, action) {
+  | ({rollbackState: RollingBack(_)}, EventBatchProcessed(_)) => (
+      {...state, currentlyProcessingBatch: false},
+      [Rollback],
+    )
```

Comment:
Yeah sure, it's a bit difficult to follow how the actions/tasks react to each other.

So `Rollback` action will first get dispatched after block range response when a reorg gets detected here:

https://github.com/enviodev/hyperindex/blob/d476b6e01702be1ed2e9da6f712683d966b4257e/codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res#L403-L414

Then on the Rollback task, if it only starts if a batch is not currently processing.

https://github.com/enviodev/hyperindex/blob/d476b6e01702be1ed2e9da6f712683d966b4257e/codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res#L1050-L1054

When the EventBatchProcessed action gets dispatched, it will be an invalid action and all we want it to do is set currently processing to false and then dispatch the rollback task again.


I just added an extra check here that rollback state will be `RollingBack` so that we don't match on this case if it were produced say by this preRegistration feature.